src/js/base/ScrollList.tsx

```diff
@@ -34,22 +47,31 @@ type ScrollListProps = {
  * @returns An infinitely scrolling list of items
  */
 
-export const ScrollList = ({ fetchNextPage, isFetchingNextPage, isLoading, items, renderRow }: ScrollListProps) => {
+export const ScrollList = ({
+    className,
+    fetchNextPage,
+    isFetchingNextPage,
+    isLoading,
+    items,
+    renderRow,
+}: ScrollListProps) => {
     useEffect(() => {
+        const scrollListElement = document.getElementById("scroll-list");
+
         const onScroll = () => {
-            if (getScrollRatio() > 0.8 && !isFetchingNextPage) {
+            if (getScrollRatio(scrollListElement) > 0.8 && !isFetchingNextPage) {
                 void fetchNextPage();
             }
         };
 
-        window.addEventListener("scroll", onScroll);
-        return () => window.removeEventListener("scroll", onScroll);
+        scrollListElement.addEventListener("scroll", onScroll);
+        return () => scrollListElement.removeEventListener("scroll", onScroll);
     }, [isFetchingNextPage, fetchNextPage]);
 
     const entries = map(items, item => renderRow(item));
 
     return (
-        <StyledScrollList>
+        <StyledScrollList id={"scroll-list"} className={className}>
```

Comment:
```suggestion
        <StyledScrollList id={"scroll-list"} className={className} onScroll={onScroll}>
```
I think this should do the same thing as your id based solution in a more reacty way. Worth checking that this soln doesn't mess with implementations where there is no height limit. (if the user scroll the whole window it might not get picked up depending on cursor position)